### PR TITLE
Add User-Agent to requests

### LINKED ISSUE #231: Set user-agent for all requests
in constants.py:

```python
USER_AGENT = "DepHell"
```

In all requests:

```python
requests.get(..., headers={'User-Agent': USER_AGENT})
```

### PR DESCRIPTION
**Pull request by hugovk**
For all `requests.get`, not only for pypistats.org.

Fixes #231.

### CODE REVIEW

**Comment by hugovk** [comment]
Any idea how to fix the Python 3.7 test failure?

**Comment by orsinium** [comment]
> Any idea how to fix the Python 3.7 test failure?

I'll take care of it, don't worry.

Could you add `def requests_session(*, auth=None, headers=None, **kwargs):` into `networking.py`? We can use it as [session context](https://2.python-requests.org/en/master/api/#request-sessions). As we do for `aiohttp`. I think it's better to keep this kind of things in one place. Another example is set timeouts for requests. What do you think?

**Comment by hugovk** [comment]
Something like this?

```diff
diff --git a/dephell/cacher.py b/dephell/cacher.py
index faaf7ea..73a8a68 100644
--- a/dephell/cacher.py
+++ b/dephell/cacher.py
@@ -18,6 +18,7 @@ from packaging.requirements import Requirement
 from dephell.constants import DEFAULT_WAREHOUSE, USER_AGENT
 from dephell.controllers import DependencyMaker
 from dephell.models import RootDependency
+from dephell.networking import requests_session
 from dephell.repositories import WarehouseAPIRepo
 
 
@@ -27,7 +28,8 @@ URL = 'https://hugovk.github.io/top-pypi-packages/top-pypi-packages-30-days.min.
 
 def get_deps():
     root = RootDependency()
-    response = requests.get(URL, headers=USER_AGENT)
+    with requests_session(headers=USER_AGENT) as session:
+        response = session.get(URL)
     for info in response.json()['rows']:
         yield from DependencyMaker.from_requirement(
             source=root,
diff --git a/dephell/networking.py b/dephell/networking.py
index 7b39244..6c49b3e 100644
--- a/dephell/networking.py
+++ b/dephell/networking.py
@@ -3,6 +3,7 @@ from ssl import create_default_context
 
 # external
 import certifi
+import requests
 from aiohttp import ClientSession, TCPConnector
 
 
@@ -16,3 +17,10 @@ def aiohttp_session(*, auth=None, **kwargs):
     except TypeError:
         connector = TCPConnector(ssl_context=ssl_context)
     return ClientSession(headers=headers, connector=connector, **kwargs)
+
+
+def requests_session(*, auth=None, headers=None, **kwargs):
+    s = requests.Session()
+    s.auth = auth
+    s.headers = headers
+    return s

```

**Comment by orsinium** [comment]
Right, but also add `User-Agent` in headers inside this session:

```python
with requests_session() as session:
  ...

def requests_session(*, auth=None, headers=None, **kwargs):
  if headers is None:
    headers = dict()
  headers.setdefault('User-Agent', USER_AGENT)
  session = requests.Session()
  session.auth = auth
  session.headers = headers
  return session
```

File: dephell/networking.py [hunk 1]
```diff
@@ -3,8 +3,14 @@
 
 # external
 import certifi
+import requests
 from aiohttp import ClientSession, TCPConnector
 
+# app
+from . import __version__
+
+USER_AGENT = {'User-Agent': 'DepHell/{version}'.format(version=__version__)}
```
Comment (orsinium) [inline dephell/networking.py]: This must be a string, not a dict.

**Comment by orsinium** [comment]
Great! Thank you, it's a big work. One small fix (I've put a comment above), and you're done :)

File: dephell/networking.py (hunk 1 above)
Comment (orsinium) [reply dephell/networking.py]: Because you're passing it as a value in headers dict.

Comment (hugovk) [reply dephell/networking.py]: Ah yes, updated!

**Comment by orsinium** [comment]
Perfect

**Comment by orsinium** [comment]
Oh, sorry. `requests.Session` doesn't accept keyword arguments. That's strange :thinking: 

```python
session = requests.Session()
if auth:
  session.auth = auth
session.headers = headers
if kwargs:
  session.__dict__.update(kwargs)
return session
```

**Comment by orsinium** [comment]
I can take care of it if you want

**Comment by hugovk** [comment]
Yes, please take over! :)

**Comment by orsinium** [comment]
I love pip releases! Another day, another broken API

**Comment by orsinium** [comment]
Thank you again :)

**Comment by hugovk** [comment]
You're welcome, and thanks for finishing it off!

**Pull request merged**